core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -530,33 +623,35 @@ boolean pointToValue(OpData opData) {
         // the straight forward helping didn't work, check why
         Operation operation = opData.op;
 
-        // the operation is remove, means we tried to change the handle index we knew about to -1
-        // the old handle index is no longer there so we have nothing to do. Note that in case of non-ZC removal the
+        // the operation is remove, means we tried to change the value reference we knew about to DELETED_VALUE
+        // the old value reference is no longer there so we have nothing to do. Note that in case of non-ZC removal the
         // loop in remove() will lookup the key again so we will still return the previous value
         if (operation == Operation.REMOVE) {
             return true; // this is a remove, no need to try again and return doesn't matter
         }
 
         // the operation is either NO_OP, PUT, PUT_IF_ABSENT, COMPUTE
-        int expectedHandleIdx = opData.handleIndex;
-        int foundHandleIdx = getEntryField(opData.entryIndex, OFFSET_HANDLE_INDEX);
+        long expectedValueReference = opData.newValueReference;
+        long foundValueReference = getValueReference(opData.entryIndex);
+        int foundValueBlockAndLength = UnsafeUtils.longToInts(foundValueReference)[BLOCK_ID_LENGTH_ARRAY_INDEX];
 
-        if (foundHandleIdx == expectedHandleIdx) {
+        if (expectedValueReference == foundValueReference) {
             return true; // someone helped
-        } else if (foundHandleIdx < 0) {
-            // the handle was deleted, retry the attach
-            opData.prevHandleIndex = -1;
+        } else if (foundValueBlockAndLength == INVALID_BLOCK_ID) {
```

Comment:
This is the check whether value was deleted. And there are multiple ways to check whether value was deleted... E.G. when we are going to really delete it later we will try to assign value reference to be DELETED_VALUE... May be at least some explanatory comment here...